src/main/java/games/strategy/util/Util.java

```diff
@@ -122,4 +129,19 @@ public static String getStringFromInputStream(InputStream in) {
     }
     return builder.toString();
   }
+
+  /**
+   * Creates a hash of the given String based on the SHA-512 algorithm.
+   * 
+   * @param input The input String to hash.
+   * @return A hashed hexadecimal String of the input.
+   */
+  public static String sha512(String input) {
+    try {
+      return String.format("%064x",
+          new BigInteger(1, MessageDigest.getInstance(SHA_512).digest(input.getBytes(StandardCharsets.UTF_8))));
```

Comment:
It might be clearer to use Guava's `BaseEncoding.base16().encode(byte[])` method here.